src/hash/DataHash.ts

```diff
@@ -46,17 +46,14 @@ export default class DataHash {
   }
 
   /**
+   * Test equality of 2 objects as DataHash.
```

Comment:
it might be better to check the deep equality of two DataHash objects instead of just checking the imprints. you might consider using a deep equality library:

https://www.npmjs.com/package/fast-deep-equal offers typed array support (`import equal from "fast-deep-equal/es6";`) and works in browsers/node

only problem is that the following test cases would fail as fast-deep-equals considers them equal, as they are equal:

    expect(DataHash.equals(undefined, undefined)).toBeFalsy();
    expect(DataHash.equals(null, null)).toBeFalsy();
    expect(DataHash.equals({}, {})).toBeFalsy();
    expect(DataHash.equals({ imprint: null }, { imprint: null })).toBeFalsy();

in addition the following test case would also fail:

    const imprint = new Uint8Array(33);
    imprint.set([1]);
    expect(
      DataHash.equals(
        DataHash.create(HashAlgorithm.SHA2_256, new Uint8Array(32)),
        { imprint }
      )
    ).toBeTruthy();